pimd/pim_bsm.c

```diff
@@ -1170,8 +1165,11 @@ static bool pim_bsm_parse_install_g2rp(struct bsm_scope *scope, uint8_t *buf,
 			return false;
 		}
 		group.prefixlen = grpinfo.group.mask;
-		group.u.prefix4.s_addr = grpinfo.group.addr.s_addr;
-
+#if PIM_IPV == 4
+		group.u.prefix4 = grp_addr;
```

Comment:
use `pim_addr_to_prefix`

Reply:
pim_addr_to_prefix will overwrite the group.prefixlen to PIM_MAX_BITLEN,  So not using.

Reply:
Just set the prefixlen after `pim_addr_to_prefix` instead of before…